tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3196,48 +3196,20 @@ public function userGetsTheFollowingPropertiesOfFileInsideSpaceUsingTheWebdavApi
 		$this->featureContext->setResponseXmlObject($responseXml);
 	}
 
-	/**
-	 * @Then /^as user "([^"]*)" (?:file|folder|entry|resource) "([^"]*)" inside space "([^"]*)" should contain a property "([^"]*)" with value "([^"]*)"$/
-	 *
-	 * @param string    $user
-	 * @param string    $resourceName
-	 * @param string    $spaceName
-	 * @param string    $property
-	 * @param string    $expectedValue
-	 *
-	 * @return void
-	 *
-	 * @throws Exception|GuzzleException
-	 */
-	public function asUserFileInsideSpaceShouldContainAPropertyWithValue(
-		string $user,
-		string $resourceName,
-		string $spaceName,
-		string $property,
-		string $expectedValue
-	):void {
-		$space = $this->getSpaceByName($user, $spaceName);
-		$fullUrl = $space["root"]["webDavUrl"] . '/' . ltrim($resourceName, "/");
-		$body = WebDavHelper::getBodyForPropfind([$property]);
-		$headers['Depth'] = '1';
-		$this->featureContext->setResponse(
-			$this->sendPropfindRequestToUrl(
-				$fullUrl,
-				$user,
-				$this->featureContext->getPasswordForUser($user),
-				'',
-				$headers,
-				$body
-			)
-		);
-		$responseXml = $this->featureContext->getResponseXml(null, __METHOD__);
-		$this->featureContext->setResponseXmlObject($responseXml);
-		$this->webDavPropertiesContext->checkSingleResponseContainsAPropertyWithValueAndAlternative(
-			$property,
-			$expectedValue,
-			$expectedValue
-		);
-	}
+    /**
+     * @Then /^as user "([^"]*)" (?:file|folder|entry) "([^"]*)" inside space "([^"]*)" (should|should not) be favorited$/
+     *
+     * @param string $user
+     * @param string $path
+     * @param string $spaceName
+     * @param string $shouldOrNot
+     *
+     * @return void
+     */
+    public function asUserFileOrFolderInsideSpaceShouldOrNotBeFavorited(string $user, string $path, string $spaceName, string $shouldOrNot):void {
+        $this->setSpaceIDByName($user, $spaceName);
+        $this->favoritesContext->asUserFileOrFolderShouldBeFavorited($user, $path, ($shouldOrNot === 'should') ? 1 : 0);
+    }
```

Comment:
```suggestion
	/**
	 * @Then /^as user "([^"]*)" (?:file|folder|entry|resource) "([^"]*)" inside space "([^"]*)" should contain a property "([^"]*)" with value "([^"]*)"$/
	 *
	 * @param string    $user
	 * @param string    $resourceName
	 * @param string    $spaceName
	 * @param string    $property
	 * @param string    $expectedValue
	 *
	 * @return void
	 *
	 * @throws Exception|GuzzleException
	 */
	public function asUserFileInsideSpaceShouldContainAPropertyWithValue(
```
Try to reuse core code for this step. If we can reuse core code in this step then it will cover many properties but if we refactor this step for only favorite property again we have to add new step for other property. In this way only size of file increases.

Reply:
@amrita-shrestha  The newly added steps basically check the step that you are suggesting. And the steps `And as user "Brian" folder "/PARENT" inside space "Shares Jail" should be favorited` makes sense after user favorites a resource rather than asserting property with value which basically the newly added step it self checks.